src/views/Timesheets/index.js

```diff
@@ -0,0 +1,539 @@
+      return jobs.getSavedShiftsForUser();
+    };
+
+    const handleDateChange1 = date => {
+      setSelectedDateIn(date);
+      handleTimeOutIsBeforeTimeIn(date, selectedDateOut);
+    };
+
+    const handleDateChange2 = date => {
+      setSelectedDateOut(date);
+      handleTimeOutIsBeforeTimeIn(selectedDateIn, date);
+    };
+
+    const handleSaveButtonClick = () => {
+      let timeIn = selectedDateIn;
+      let timeOut = selectedDateOut;
+
+      if (selectedDateIn.getDay() === 6 && selectedDateOut.getDay() === 0) {
+        let timeOut2 = new Date(timeOut.getTime());
+        let timeIn2 = new Date(timeOut.getTime());
+        timeIn2.setHours(0);
+        timeIn2.setMinutes(0);
+        timeIn2.setSeconds(0);
+        timeIn2.setMilliseconds(0);
+
+        timeOut.setDate(timeIn.getDate());
+        timeOut.setHours(23);
+        timeOut.setMinutes(59);
+        timeOut.setSeconds(0);
+        timeOut.setMilliseconds(0);
+
+        let timeDiff2 = timeOut2.getTime() - timeIn2.getTime();
+        let calculatedTimeDiff2 = timeDiff2 / 1000 / 60 / 60;
+        let roundedHourDifference2 = (Math.round(calculatedTimeDiff2 * 4) / 4).toFixed(2);
+
+        saveShift(
+          selectedJob.EMLID,
+          timeIn2.toLocaleString(),
+          timeOut2.toLocaleString(),
+          roundedHourDifference2,
+          userShiftNotes,
+        ).then(result => {
+          setSnackbarOpen(true);
+        });
+      }
+
+      let timeDiff1 = timeOut.getTime() - timeIn.getTime();
+      let calculatedTimeDiff = timeDiff1 / 1000 / 60 / 60;
+      let roundedHourDifference = (Math.round(calculatedTimeDiff * 4) / 4).toFixed(2);
+
+      saveShift(
+        selectedJob.EMLID,
+        timeIn.toLocaleString(),
+        timeOut.toLocaleString(),
+        roundedHourDifference,
+        userShiftNotes,
+      ).then(result => {
+        shiftDisplayComponent.loadShifts()
+        setSelectedDateOut(null);
+        setSelectedDateIn(null);
+        setUserShiftNotes('');
+        setUserJobs([]);
+        setHoursWorkedInDecimal(0);
+      }).catch(err => {
+        if (err.toLowerCase().includes('overlap')) {
+          setIsOverlappingShift(true);
+        }
+      });
+    };
+
+    const saveShift = async (
+      eml,
+      shiftStart,
+      shiftEnd,
+      hoursWorked,
+      shiftNotes,
+    ) => {
+      await jobs.saveShiftForUser(
+        eml,
+        shiftStart,
+        shiftEnd,
+        hoursWorked,
+        shiftNotes,
+      );
+    };
+
+    const jobsMenuItems = userJobs ? (
+      userJobs.map(job => (
+        <MenuItem label={job.POSTITLE} value={job} key={job.EMLID}>
+          {job.POSTITLE}
+        </MenuItem>
+      ))
+    ) : (
+        <></>
+      );
+
+    const isLeapYear = date => {
+      if (date.getFullYear() % 4 === 0) {
+        if (date.getFullYear() % 100 === 0) {
+          if (date.getFullYear() % 400 !== 0) {
+            return 'false';
+          }
+          if (date.getFullYear() % 400 === 0) {
```

Comment:
Could 197, 201, and 205 just be `else`?

Reply:
Probably. Because it's so finicky I'm kind of afraid to touch it, but we will try it anyway